src/plan/mutator_context.rs

```diff
@@ -78,6 +80,48 @@ impl<VM: VMBinding> std::fmt::Debug for MutatorConfig<VM> {
     }
 }
 
+/// Used to build a mutator struct
+pub struct MutatorBuilder<VM: VMBinding> {
+    allocators: Allocators<VM>,
```

Comment:
I think we can do this:
```rust
pub struct MutatorBuilder<VM: VMBinding> {
    barrier: Box<dyn Barrier<VM>>,
    /// The mutator thread that is bound with this Mutator struct.
    mutator_tls: VMMutatorThread,
    mmtk: &'static MMTK<VM>,
    config: MutatorConfig<VM>,
}
```

The constructing of `Allocators` will be done when we call `build()`.
```rust
    pub fn build(self) -> Mutator<VM> {
        Mutator {
            allocators: Allocators::<VM>::new(self.mutator_tls, self.mmtk, &self.config.space_mapping),
            barrier: self.barrier,
            mutator_tls: self.mutator_tls,
            plan: self.mmtk.get_plan(),
            config: self.config,
        }
    }
```